diff --git a/driver/smoothie_driver.py b/driver/smoothie_driver.py
index fae6dcc..8732152 100644
--- a/driver/smoothie_driver.py
+++ b/driver/smoothie_driver.py
@@ -4,6 +4,7 @@
 import asyncio, json, copy
 import datetime
 import sys
+import os
 from collections import Callable
 
 
@@ -392,7 +393,12 @@ class SmoothieDriver(object):
 				#yield from server.wait_closed()
 				print('here')
 			else:
-				asyncio.async(self.the_loop.create_connection(lambda: callbacker, host='0.0.0.0', port=3333))
+				smoothie_host=os.environ.get('SMOOTHIE_HOST', '0.0.0.0')
+				smoothie_port=int(os.environ.get('SMOOTHIE_PORT', '3333'))          
+				asyncio.async(self.the_loop.create_connection(
+		  			lambda: callbacker, 
+		  			host=smoothie_host, 
+		  			port=smoothie_port))
 		
 		except:
 			print(datetime.datetime.now(),' - error:driver.connects\n\r',sys.exc_info())
