--- communication.js
+++ communication-new.js
@@ -62,7 +62,7 @@
 		wsuri = "ws://127.0.0.1:8080/ws";
 	} else {
     	//wsuri = (document.location.protocol === "http:" ? "ws:" : "wss:") + "//0.0.0.0:8080/ws";
-    	wsuri = (document.location.protocol === "http:" ? "ws:" : "wss:") + "//10.10.1.2:8080/ws";
+    	wsuri = (document.location.protocol === "http:" ? "ws:" : "wss:") + "//otone.local:8080/ws";
     	console.log("\twsuri: " + wsuri);
     	//document.location.host + "/ws";
 	}
